pilot/pkg/xds/cds.go

```diff
@@ -64,38 +64,15 @@ var pushCdsGatewayConfig = func() sets.Set[kind.Kind] {
 }()
 
 func cdsNeedsPush(req *model.PushRequest, proxy *model.Proxy) bool {
-	if proxy.Type == model.Ztunnel {
-		// Not supported for ztunnel
-		return false
+	if res, ok := xdsNeedsPush(req, proxy); ok {
+		return res
 	}
-	if req == nil {
+	if waypointNeedsPush(req) {
```

Comment:
donot you need to check this is waypoint?

Reply:
We could. But it is redundant I think. Since Address is only relevant for Waypoint. Any way will change